.github/workflows/generate_chapters.yml

```diff
@@ -0,0 +1,43 @@
+#Runs "npm run generate" on any pushes to master for markdown, templates or generate scripts
+#Then we commit any template changes generated by the "npm run generate"
+#
+#We do it on a push, rather than on pull request, and only on master to avoid merge conflicts
+name: Generate Chapters on Push
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - src/content/**
+      - src/templates/**
+      - tools/generate/**
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Setup Node.js for use with actions
+      uses: actions/setup-node@v1.1.0
+      with:
+        version:  12.x
+    - name: Checkout branch
+      uses: actions/checkout@v2
+    - name: Generate the chapters
+      run: |
+        echo "regenerating chapters"
+        cd src
+        npm install
+        echo "regenerating chapters"
+        npm run generate
+    - name: Commit changes
+      run: |
+        echo "Checking git status"
+        git status
+        echo "Adding templates and content directory"
+        git config --global user.email "${GITHUB_ACTOR}@users.noreply.github.com"
+        git config --global user.name "Automatic Update"
+        git add src/templates/*/*/chapters/
+        git add src/templates/sitemap.xml
+        echo "Commiting changes"
+        git diff-index --quiet HEAD || git commit -m "Regenerated chapters"
+        git push
```

Comment:
Does it have to push to master? Is it possible to push to a working branch and automatically open a housekeeping PR? I've seen auto-PRs done in the past with bots that patch security holes, eg https://github.com/HTTPArchive/httparchive.org/pull/183

Reply:
I've been [chatting to one of the maintainers of the GitHub checkout action on this](https://github.com/actions/checkout/issues/124#issuecomment-580018424) and it's not currently possible. The issue is open so am definitely up for changing this in future if it becomes possible.

I did look at how Calibre does this ([discussed in that issue](https://github.com/actions/checkout/issues/124#issuecomment-584384687)) but it looks complex using the GitHub API rather than direct actions themselves.

So basically yes, to keep it simple, it has to be done straight into master.

Personally I think for the moment the fact that we have limited it to the auto-generated content (as discussed above) with these two lines:
```
        git add src/templates/*/*/chapters/
        git add src/templates/sitemap.xml
```
means the risk is low. Worst case if master gets in a bad state we disable this, and just run `npm run generate` and fix it with a new PR.

Ideally I would like to auto generate the timestamps to fix #317 as well, and I have that working too, but that would mean touching the non-generated content (the md files and the static htmls) so not comfortable doing this until GitHub actions provides some way to add this to PRs instead of straight into master. So I have not included this in there now.

Reply:
I'm ok with this if you're confident that it won't blow up `master` and you're prepared to handle any bugs :)